src/DataTable/selection/tests/ControlledSelect.test.jsx

```diff
@@ -30,29 +30,45 @@ describe('<ControlledSelect />', () => {
   beforeEach(() => {
     jest.resetAllMocks();
   });
-  it('correctly selects a row', () => {
+
+  it('correctly selects a row', async () => {
     const isChecked = true;
     mockGetToggleRowSelectedProps.mockReturnValue({ checked: isChecked });
     const spy = jest.spyOn(selectActions, 'addSelectedRowAction');
     const row = { ...baseRow, isSelected: false };
     const selectProps = { row };
-    const wrapper = mount(
-      <ControlledSelectWrapper tableProps={tableProps} selectProps={selectProps} />,
+    render(
+      <ControlledSelectWrapper
+        tableProps={tableProps}
+        selectProps={selectProps}
+        data-testid="select-checkbox"
+      />,
     );
-    toggleCheckbox({ isChecked, wrapper });
+
+    const checkbox = screen.getByTestId('select-checkbox');
```

Comment:
It would be better to follow `react-testing-library` philosophy and find the element by its accessible name, e.g. `const checkbox = screen.getByRole('checkbox');`, then you wouldn't need to add `data-testid` everywhere in this file.